### ISSUE TITLE
`check` should not consider 503s as fatal

### ISSUE THREAD

**Issue by olix0r** [linkerd/linkerd2#1851]
When installing linkerd, the `check` command will fail (and not retry) if the API returns a 503.  This error is routinely returned when the linkerd namespace hasn't been fully provisioned (i.e and we're waiting for the services to become available).

```
kubernetes-api: can initialize the client..................................[ok]
kubernetes-api: can query the Kubernetes API...............................[ok]
kubernetes-api: is running the minimum Kubernetes API version..............[ok]
linkerd-api: control plane namespace exists................................[ok]
linkerd-api: control plane pods are ready..................................[ok]
linkerd-api: can initialize the client.....................................[ok]
linkerd-api: can query the control plane API...............................[FAIL] -- Unexpected API response: 503 Service Unavailable

Status check results are [FAIL]
```

**Comment by klingerf** [comment]
@olix0r Ah, interesting -- the check code already retries until all pods have passed all readiness checks, but I guess it can take even longer than that to provision services in some setups.

@alenkacz I think the right fix here is to just mark that particular check as retryable, so that it will be retried automatically if the `--wait` duration hasn't been exceeded.

**Comment by olix0r** [comment]
I think I'm still seeing the issue I originally reported.

Running a CLI based on edge-18.11.2

```namespace "ver-linkerd" configured
serviceaccount "linkerd-controller" unchanged
clusterrole "linkerd-ver-linkerd-controller" configured
clusterrolebinding "linkerd-ver-linkerd-controller" configured
serviceaccount "linkerd-prometheus" unchanged
clusterrole "linkerd-ver-linkerd-prometheus" configured
clusterrolebinding "linkerd-ver-linkerd-prometheus" configured
service "api" configured
service "proxy-api" configured
deployment "controller" configured
customresourcedefinition "serviceprofiles.linkerd.io" configured
service "web" configured
deployment "web" configured
service "prometheus" configured
deployment "prometheus" configured
configmap "prometheus-config" configured
service "grafana" configured
deployment "grafana" configured
configmap "grafana-config" configured
kubernetes-api: can initialize the client..................................[ok]
kubernetes-api: can query the Kubernetes API...............................[ok]
kubernetes-api: is running the minimum Kubernetes API version..............[ok]
linkerd-api: control plane namespace exists................................[ok]
linkerd-api: control plane pods are ready..................................[ok]
linkerd-api: can initialize the client.....................................[ok]
linkerd-api: can query the control plane API...............................[FAIL] -- Unexpected API response: 503 Service Unavailable

Status check results are [FAIL]
```

**Comment by klingerf** [comment]
@olix0r I think this is a result of the readiness probe changes that were made in #1899. They're being rolled back in #1912, which should fix this issue. See #1919 for more discussion about what's going on.

**Comment by klingerf** [comment]
This should be fixed now that #1912 is merged, but please reopen if you're still seeing it.
